llama_index/agent/openai_agent.py

```diff
@@ -57,282 +63,226 @@ def call_function(
     )
 
 
-class BaseOpenAIAgent(BaseAgent):
-    """Base OpenAI Agent."""
+class ChatSession:
+    def __init__(
+        self,
+        memory: BaseMemory,
+        prefix_messages: List[ChatMessage],
+        get_tools_callback: Callable[[str], List[BaseTool]],
+    ):
+        self.memory = memory
+        self.prefix_messages = prefix_messages
+        self.get_tools_callback = get_tools_callback
```

Comment:
nit: I find it quite peculiar for a chat session to have a `get_tools_callback`.

Reply:
Agreed, that's got a bad code smell. Removed `ChatSession` entirely. It was a thin wrapper for memory anyway.